File: infrastructure/resources/models/status-change-event.ts
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: MIT-0

import { Payload } from "./payload"
import { Status } from "./status";

export class StatusChangeEvent extends Payload {
    constructor(init?:Partial<StatusChangeEvent>) {
        super("StatusChangeEvent");
        Object.assign(this, init);
    } 
    public userId!: string;
    public currentStatus!: Status;
    public eventDate!: Date;
}

File: infrastructure-net7/src/Shared/Models/StatusChangeEvent.cs
namespace Shared.Models;

public class StatusChangeEvent : Payload
{
    public StatusChangeEvent(string type) : base(type)
    {
    }

    public StatusChangeEvent(string userId, Status currentStatus, DateTime eventDate) : base("StatusChangeEvent")
    {
        this.userId = userId;
        this.currentStatus = currentStatus;
        this.eventDate = eventDate;
    }

    public string? userId { get; set; }
    public Status? currentStatus { get; set; }
    public DateTime? eventDate { get; set; }
}